charon/src/pretty/fmt_with_ctx.rs

```diff
@@ -14,11 +14,41 @@ use crate::{
 };
 use hax_frontend_exporter as hax;
 
-impl BlockData {
-    pub fn fmt_with_ctx<C>(&self, tab: &str, ctx: &C) -> String
-    where
-        C: AstFormatter,
-    {
+/// Format the AST type as a string.
+pub trait FmtWithCtx<C> {
+    fn fmt_with_ctx(&self, ctx: &C) -> String {
+        // By default use no tab.
+        self.fmt_with_ctx_and_indent("", ctx)
+    }
+
+    fn fmt_with_ctx_and_indent(&self, _tab: &str, _ctx: &C) -> String {
+        panic!("This type does not know how to be formatted with indent")
+    }
+
+    /// Returns a struct that implements `Display`. This allows the following:
+    ///     println!("{}", self.with_ctx(ctx));
```

Comment:
```suggestion
    /// ```text
    ///     println!("{}", self.with_ctx(ctx));
    /// ```
```